From: Jack Morgenstein <jackm@dev.mellanox.co.il>
Date: Wed, 30 May 2012 09:14:50 +0000
Subject: net/mlx4_core: Fix the slave_id out-of-range test in mlx4_eq_int
Patch-mainline: v3.5-rc1
Git-commit: 30f7c73bed749814f4985b1ce4566fe64b9c25f0
References: bnc#786036 FATE#314304

Ths fixes the comparison in the FLR (Function Level Reset) event case.

Signed-off-by: Jack Morgenstein <jackm@dev.mellanox.co.il>
Reviewed-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/mellanox/mlx4/eq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
--- a/drivers/net/ethernet/mellanox/mlx4/eq.c
+++ b/drivers/net/ethernet/mellanox/mlx4/eq.c
@@ -425,7 +425,7 @@ static int mlx4_eq_int(struct mlx4_dev *
 
 			mlx4_dbg(dev, "FLR event for slave: %d\n", flr_slave);
 
-			if (flr_slave > dev->num_slaves) {
+			if (flr_slave >= dev->num_slaves) {
 				mlx4_warn(dev,
 					  "Got FLR for unknown function: %d\n",
 					  flr_slave);
